app/code/core/Mage/Eav/controllers/Adminhtml/Attribute/AbstractController.php

```diff
@@ -0,0 +1,337 @@
+<?php
+/**
+ * OpenMage
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@magento.com so we can send you a copy immediately.
+ *
+ * @category    Mage
+ * @package     Mage_Eav
+ * @copyright  Copyright (c) 2006-2020 Magento, Inc. (http://www.magento.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+/**
+ * Attribute controller
+ *
+ * @category   Mage
+ * @package    Mage_Eav
+ * @author      Magento Core Team <core@magentocommerce.com>
+ */
+
+abstract class Mage_Eav_Adminhtml_Attribute_AbstractController extends Mage_Adminhtml_Controller_Action
```

Comment:
For consistency, is it better to place this in `app/code/core/Mage/Adminhtml/Controller/Eav/Abstract.php`? The code in `Block/Adminhtml` would also need to be placed in `Mage/Adminhtml/Eav`.

Reply:
I debated this myself. I recall a discussion somewhere on this GitHub project about how Magento team stuffed everything into Adminhtml because the old code didn't have support for defining adminhtml controllers in the specific module. Magento team later added that support but never broke out the controllers / blocks to the correct module. This was a cleanup project so I didn't want to add to the mess.

Reply:
We also already have a directory at `/app/code/core/Mage/Eav/Block/Adminhtml`

Reply:
Thanks for the clarification, I wasn't aware of the history. However I haven't seen abstract class in the `controller` folder. How about placing it in 
`app/code/core/Mage/Eav/Controller/Adminhtml/Attribute/Abstract.php`

Reply:
Done, thanks for the suggestion